### PR TITLE
Remove worker/pool client

### PR DESCRIPTION
**Pull request by clarafu**
The worker.Client interface only existed because there were parts of the
code that wanted to choose a worker and then find or create a container
but there were other parts that wanted to use a specific worker and then
find or create container on that worker. There were these two paths but
in order to use the same code, both the pool and worker objects were
adjusted in order to fit this worker Client interface. This commit
removes this dependency by having the pool object find or choose a
worker and then on the worker object find or create a container on that
chosen worker.

Also, get containers are now being placed with the chosen placement
strategy instead of always randomly placing them.

### CODE REVIEW

**Comment by cirocosta** [comment]
Hey @clarafu 

Is this PR still a thing? I couldn't immediately find any other issue that this one is blocked on 🤔

Thanks!

**Comment by vito** [comment]
It's done, I think we just wanted to run TopGun against it and wait until 5.0 before merging.

File: atc/atccmd/command.go [hunk 1]
```diff
@@ -500,13 +500,16 @@ func (cmd *RunCommand) constructAPIMembers(
...
 	}
 
+	resourceFactory := resource.NewResourceFactory()
 	dbResourceCacheFactory := db.NewResourceCacheFactory(dbConn, lockFactory)
-	resourceFetcherFactory := resource.NewFetcherFactory(lockFactory, clock.NewClock(), dbResourceCacheFactory)
+	fetchSourceFactory := resource.NewFetchSourceFactory(dbResourceCacheFactory, resourceFactory)
+	resourceFetcher := resource.NewFetcher(clock.NewClock(), lockFactory, fetchSourceFactory)
 	dbResourceConfigFactory := db.NewResourceConfigFactory(dbConn, lockFactory)
 	imageResourceFetcherFactory := image.NewImageResourceFetcherFactory(
-		resourceFetcherFactory,
```
Comment (topherbullock) [inline atc/atccmd/command.go]: YES! Get outta here `FetcherFactory`!

File: atc/atccmd/command.go [hunk 2]
```diff
@@ -500,13 +500,16 @@ func (cmd *RunCommand) constructAPIMembers(
 		return nil, err
 	}
 
+	resourceFactory := resource.NewResourceFactory()
 	dbResourceCacheFactory := db.NewResourceCacheFactory(dbConn, lockFactory)
-	resourceFetcherFactory := resource.NewFetcherFactory(lockFactory, clock.NewClock(), dbResourceCacheFactory)
+	fetchSourceFactory := resource.NewFetchSourceFactory(dbResourceCacheFactory, resourceFactory)
```
Comment (topherbullock) [inline atc/atccmd/command.go]: Looks like the behavior in `FetchSource` could be collapsed into `Fetcher`:
- we only ever instantiate them in `atccmd`
- `FetchSource`'s interface is only relied on by `Fetcher`

That would be.. 
SO FETCH

Comment (topherbullock) [reply atc/atccmd/command.go]: Might also be feasible to roll all of this up into the current resource factory, as its only there as a boundary for tests.. but I wouldn't block merging this on that change!

**Comment by ddadlani** [comment]
Agreed with @topherbullock - I don't like how the `resource` package looks right now but that can be refactored later. Looks good otherwise!

**Pull request merged**